File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq

def solution(n, works):
    answer = 0
    if sum(works) <= n:
        return 0
    works = [-work for work in works]
    heapq.heapify(works)
    while n > 0:
        maxVal = heapq.heappop(works)
        heapq.heappush(works, maxVal + 1)
        n -= 1
    for work in works:
        answer += work**2
    
    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.js
function solution(n, works) {
  if (works.reduce((a, b) => a + b, 0) <= n) return 0;

  let sorted = works.sort((a, b) => b - a);

  while (n) {
    let max_num = sorted[0];
    
    for (let i = 0; i < sorted.length; i++) {
      if (sorted[i] >= max_num) {
        sorted[i] -= 1;
        n -= 1;
      }
      if (!n) break;
    }
  }

  return works.reduce((a, b) => a + b ** 2, 0);
}
